doc/compiler/control/CompilationControl.md

```diff
@@ -0,0 +1,304 @@
+threads created depends on the resources available on the machine,
+as well as the size of the Compilation Queue.
+
+In general, the Compilation Threads are synchronized using
+the [Compilation Monitor](#compilation-monitor). However, each
+thread also has a Compilation Thread Monitor. This monitor is
+used to synchronize per thread information, such as its State,
+as well as to suspend and resume as needed.
+
+
+## Compilation Monitor
+
+The Compilation Monitor is a global monitor that is used to
+synchronize many aspects of the compilation infrastructure.
+
+
+## Sampler Thread
+
+The Sampler Thread can be though of as a daemon within the JIT. It
+performs several checks periodically. Most notably, it is responsible
+for triggering sampling via Async Checkpoint Sampling.
+
+### Async Checkpoint Sampling
+
+Async Checkpoints, also known as Safe Points and Yield Points, are
+well defined points during a Java Thread's execution wherein the
+thread yields to the VM for the purpose of doing GC or any other
+task that requires the thread to be stopped. Async Checkpoint Sampling
+piggybacks off this infrastructure to obtain sampling information
+that is used to determine which methods should be compiled and
+what what optimization level.
+
+This is the general procedure:
+
+1. Every 10ms, the Sampler Thread sets a flag in the J9VMThead of all Java Threads.
+2. At an Async Checkpoint, the VM checks if the flag has been set, and if so, calls a JIT hook.
+3. The hook collects a bunch of information, including the Java Methods on the stack.
+
+Async Checkpoint Sampling is used for [DLT](#dynamic-loop-transfer),
+[Recompilation](#recompilation), and printing tracepoints
+containing information about the methods currently on the Java
+Thread's stack.
+
+
+## Code And Data Cache Management
+
+The compiled bodies are stored into the Code Cache; data related to the
+compiled body is stored in the Data Cache. Space in the Code and Data
+Caches have to be allocated on an as-needed basis. Additionally,
+these allocations have to be synchronized since multiple Compilation
+Threads can be active simultaneously. As such, each Compilation Thread,
+as part of the compilation, will reserve a Code Cache to ensure that
+no other thread can write to it.
+
+There unfortunately exists a bit of name overloading. When speaking
+generally about a place to store compiled bodies, the term "Code Cache"
+is used. However, the actual chunk of memory is known as the "Code
+Respository", and individual subsections of the Code Repository are
+called "Code Caches". By default, the JIT allocates a 256 MiB Code
+Repository. This is then carved into 2 MiB Code Caches. It is this
+2 MiB subsection of the Code Reponsitory that a Compilation Thread
+reserves during a compilation.
+
+### Trampolines
+
+A trampolines is essentially a snippet of code that is used branch to
+a location far from the current Instruction Pointer. Each Code Cache
```

Comment:
The Trampoline section is not reserved if the JVM is configured such that it is guaranteed that far jumps will never be needed. This is the purpose of the code cache repository: to ensure that code caches are close together and obviate the need for  far jumps.